m4/check_pkg_nvtx.m4

```diff
@@ -0,0 +1,46 @@
+# -*- autoconf -*-
+#
+# Copyright (c) 2024      Amazon.com, Inc. or its affiliates. All rights reserved.
+#
+# See LICENSE.txt for license information
+#
+
+AC_DEFUN([CHECK_PKG_NVTX], [
+  check_pkg_found=yes
+
+  check_pkg_CPPFLAGS_save="${CPPFLAGS}"
+  check_pkg_LDFLAGS_save="${LDFLAGS}"
+  check_pkg_LIBS_save="${LIBS}"
+
+  AC_ARG_WITH([nvtx],
+     [AS_HELP_STRING([--with-nvtx=DIR], [Enable tracing capability with NVTX @<:@default=no@:>@])])
+
+  AS_IF([test -z "${with_nvtx}" -o "${with_nvtx}" = "yes"],
```

Comment:
I don't speak m4, but I can't tell from here whether you're really defaulting `with_nvtx` to no. It looks like it will enable nvtx if found, whether or not the user passes `--with-nvtx`.